File: app/src/main/java/com/example/user/biblejournal/model/asynctasks/GetNoteByIdAsyncTask.java
package com.example.user.biblejournal.model.asynctasks;

import android.os.AsyncTask;

import com.example.user.biblejournal.model.Repository;
import com.example.user.biblejournal.model.database.note.NoteDao;
import com.example.user.biblejournal.model.database.note.NoteEntity;

public class GetNoteByIdAsyncTask extends AsyncTask<Integer, Void, NoteEntity> {

    private NoteDao mAsyncTaskDao;
    private Repository.NotesRepositoryListener notesRepositoryListener;

    public GetNoteByIdAsyncTask(NoteDao dao, Repository.NotesRepositoryListener notesRepositoryListener) {
        mAsyncTaskDao = dao;
        this.notesRepositoryListener = notesRepositoryListener;
    }

    @Override
    protected NoteEntity doInBackground(final Integer... params) {
        return mAsyncTaskDao.getNoteById(params[0]);
    }

    @Override
    protected void onPostExecute(NoteEntity noteEntity) {
        super.onPostExecute(noteEntity);

        notesRepositoryListener.onNoteFetchedById(noteEntity);
    }
}

File: app/src/main/java/com/example/user/biblejournal/model/asynctasks/GetNoteByIdAsyncTask.kt
package com.example.user.biblejournal.model.asynctasks

import android.os.AsyncTask

import com.example.user.biblejournal.model.Repository
import com.example.user.biblejournal.model.database.note.NoteDao
import com.example.user.biblejournal.model.database.note.NoteEntity

class GetNoteByIdAsyncTask(private val mAsyncTaskDao: NoteDao, private val notesRepositoryListener: Repository.NotesRepositoryListener) : AsyncTask<Int, Void, NoteEntity>() {

    override fun doInBackground(vararg params: Int?): NoteEntity {
        return mAsyncTaskDao.getNoteById(params[0]!!)
    }

    override fun onPostExecute(noteEntity: NoteEntity) {
        super.onPostExecute(noteEntity)

        notesRepositoryListener.onNoteFetchedById(noteEntity)
    }
}
